### PR TITLE
Use init strides

### PR DESCRIPTION
**Pull request by bensander**
Add assembly UseInitialStrides, plus some debug modes and tests

### CODE REVIEW

File: Tensile/Tests/nightly/use_initial_strides/test_2.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# benchmark assembly and source kernels
+GlobalParameters:
+  MinimumRequiredVersion: 4.2.0
+  CMakeBuildType: Debug
```
Comment (yoichiyoshida) [inline Tensile/Tests/nightly/use_initial_strides/test_2.yaml]: Should nightly tests be in debug?

File: Tensile/Tests/nightly/use_initial_strides/test_strides.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# benchmark assembly and source kernels
+GlobalParameters:
+  MinimumRequiredVersion: 4.2.0
+  CMakeBuildType: Debug
```
Comment (yoichiyoshida) [inline Tensile/Tests/nightly/use_initial_strides/test_strides.yaml]: Should nightly tests be in debug?

File: Tensile/Tests/nightly/use_initial_strides/test_strides1.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# benchmark assembly and source kernels
+GlobalParameters:
+  MinimumRequiredVersion: 4.2.0
+  CMakeBuildType: Debug
```
Comment (yoichiyoshida) [inline Tensile/Tests/nightly/use_initial_strides/test_strides1.yaml]: Should nightly tests be in debug?

File: Tensile/Tests/nightly/pack_tensor_dims/multi_free_batch.yaml [hunk 4]
```diff
@@ -45,7 +44,7 @@ BenchmarkProblems:
         - LoopTail: [True]
         - EdgeType: ["ShiftPtr"]
       ForkParameters:
-        #- KernelLanguage: ["Source","Assembly"]
+        - KernelLanguage: ["Source"]
```
Comment (yoichiyoshida) [inline Tensile/Tests/nightly/pack_tensor_dims/multi_free_batch.yaml]: Is this no longer supported in assembly?

File: Tensile/Common.py [hunk 5]
```diff
@@ -858,9 +863,10 @@
     "IndexAssignmentsA":        [0, 2],
     "IndexAssignmentsB":        [1, 2],
     "NumIndicesC":              2,
-    "UseInitialStrides":        False,
+    "UseInitialStrides":        0,  # 0x1=use initial strides for AB, 0x2=use initial strides for CD, 0x3=use initial strides for all
```
[the remark this answers is not in the document]
Comment (bensander) [reply Tensile/Common.py]: Good point.  Moved to two YAML parms: UseInitialStrideAB and UseInitialStridesCD

**Pull request merged**